src/uct/ib/base/ib_log.c

```diff
@@ -19,17 +19,29 @@ void uct_ib_log_dump_opcode(uint32_t qp_num, unsigned idx, uct_ib_opcode_t *op,
 void uct_ib_log_dump_sg_list(uct_ib_iface_t *iface, uct_am_trace_type_t type,
                              struct ibv_sge *sg_list, int num_sge,
                              uint64_t inline_bitmap,
-                             uct_log_data_dump_func_t data_dump,
+                             uct_log_data_dump_func_t data_dump_cb,
                              char *buf, size_t max)
 {
     char data[256];
+    int data_dump = (data_dump_cb != NULL);
     size_t total_len       = 0;
     size_t total_valid_len = 0;;
     char *s    = buf;
     char *ends = buf + max;
     void *md   = data;
     size_t len;
     int i;
+    uct_md_attr_t md_attr;
+    ucs_status_t status;
+
+    if (data_dump) {
```

Comment:
need to see how we can at least dump the non-zcopy SGEs (e.g always dump bcopy and inline, and dump am_zcopy headers)

Reply:
@bureddy this comment was not addressed..